File: ios/Synapse/Models/Progress.swift
import Foundation

struct TopicMastery: Codable, Identifiable, Sendable, Hashable {
    let topicId: String
    let topicName: String
    let masteryLevel: String
    let theta: Double?
    let totalResponses: Int
    let correctResponses: Int
    let accuracyRate: Double?

    var id: String { topicId }
}

struct SessionBrief: Codable, Identifiable, Sendable, Hashable {
    let id: String
    let sessionType: String
    let status: String
    let itemsDelivered: Int
    let itemsCorrect: Int
    let scorePercent: Double?
    let netScore: Double?
    let startedAt: Date
    let completedAt: Date?
}

struct ProgressOut: Codable, Sendable, Hashable {
    let globalTheta: Double?
    let globalThetaSe: Double?
    let totalResponses: Int
    let totalCorrect: Int
    let accuracy: Double?
    let topics: [TopicMastery]
    let recentSessions: [SessionBrief]
}

File: frontend/lib/types/progress.ts
// Mirrors backend app/schemas/progress.py

export interface TopicMastery {
  topic_id: string;
  topic_name: string;
  mastery_level: string;
  theta: number | null;
  total_responses: number;
  correct_responses: number;
  accuracy_rate: number | null;
}

export interface SessionBrief {
  id: string;
  session_type: string;
  status: string;
  items_delivered: number;
  items_correct: number;
  score_percent: number | null;
  net_score: number | null;
  started_at: string;
  completed_at: string | null;
}

export interface Progress {
  global_theta: number | null;
  global_theta_se: number | null;
  total_responses: number;
  total_correct: number;
  accuracy: number | null;
  topics: TopicMastery[];
  recent_sessions: SessionBrief[];
}
